### ISSUE TITLE
Properly deal with r = 0. pairs

### ISSUE THREAD

**Issue by adematti** [cosmodesi/pycorr#7]
Counting of r = 0. pairs is not very clear in Corrfunc.
One definition would be to *not* count pairs of a same object. This means that different results will be obtained for an auto or a cross-correlation of the same catalog. It would also mean tracking down the particle numbers, even in Corrfunc (which would be somewhat a pain)...
Another definition would be to *not* count zero-separation pairs (whether there are the same objects or not). Same results will be obtained for an auto or a cross-correlation of the same catalog. No tracking necessary, but maybe in case of single precision, real pairs will have r = 0. and therefore be discarded?
@ashleyjross any opinion?

**Comment by ashleyjross** [comment]
Probably the easiest thing to do is reset the minimum edge to something like 0.0001 mpc/h...perhaps set it as an option so if someone wants to test some very small binning they have the option to change?

**Comment by adematti** [comment]
Well, I think one would avoid setting some unnecessary scale.
I eventually chose the first option:
- Pairs at separation :math:`s = 0` are included in the :math:`\mu = 0` bin.
- In case of auto-correlation (no ``positions2`` provided), auto-pairs (pairs of same objects) are not counted.
- In case of cross-correlation, all pairs are counted.
- In any case, duplicate objects will be counted (with separation zero).
This required updating Corrfunc quite a bit to properly add auto-pairs in rppi/smu cases (https://github.com/adematti/Corrfunc/commit/67581af6bfbd5664b5715c67de653ee5d8533ec9), but at least it's done!
Closing.
